tools/filesystem/archivefilesystem_test.go

```diff
@@ -0,0 +1,88 @@
+package filesystem_test
+
+import (
+	"archive/tar"
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"testing"
+
+	"github.com/mandelsoft/vfs/pkg/memoryfs"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/kyma-project/modulectl/tools/filesystem"
+)
+
+func TestGenerateTarData(t *testing.T) {
+	t.Run("should generate tar data successfully", func(t *testing.T) {
+		// given
+		expectedData := []byte("Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.")
+
+		memFs := memoryfs.New()
+		err := memFs.MkdirAll("test/path", 0o755)
+		require.NoError(t, err)
+		inputFile, err := memFs.Create("test/path/file.txt")
+		require.NoError(t, err)
+		_, err = inputFile.Write(expectedData)
+		require.NoError(t, err)
+		err = inputFile.Close()
+		require.NoError(t, err)
+
+		// when
+		tarData, err := filesystem.GenerateTarData(memFs, "test/path/file.txt")
```

Comment:
Good thing we start testing this.
Let's add some other paths besides happy path to define expected behaviour of this `archivefilesystem`?